internal/flypg/node.go

```diff
@@ -350,10 +370,32 @@ func (n *Node) PostInit(ctx context.Context) error {
 				return fmt.Errorf("failed to issue registration certificate: %s", err)
 			}
 		} else {
-			// Configure as standby
-			log.Println("Registering standby")
-			if err := n.RepMgr.registerStandby(); err != nil {
-				return fmt.Errorf("failed to register new standby: %s", err)
+			if n.Witness {
+				log.Println("Registering witness")
+
+				// Create required users
+				if err := n.setupCredentials(ctx, conn); err != nil {
+					return fmt.Errorf("failed to create required users: %s", err)
+				}
+
+				// Setup repmgr database and extension
+				if err := n.RepMgr.enable(ctx, conn); err != nil {
+					return fmt.Errorf("failed to enable repmgr: %s", err)
+				}
+
+				primary, err := n.RepMgr.ResolveMemberOverDNS(ctx)
+				if err != nil {
+					return fmt.Errorf("failed to resolve primary member: %s", err)
+				}
+
+				if err := n.RepMgr.registerWitness(primary.Hostname); err != nil {
+					return fmt.Errorf("failed to register witness: %s", err)
+				}
+			} else {
+				log.Println("Registering standby")
+				if err := n.RepMgr.registerStandby(); err != nil {
+					return fmt.Errorf("failed to register new standby: %s", err)
```

Comment:
I've been thinking for a while we should set up a better logging system (obviously doesn't have to happen now) so we can annotate logs with the codepath they are hitting. Like instead of just loging:
```
failed to register new standby: %s
```
we could do something like
```
[witness-registration] failed to register new standby: %s
```
since that error can possibly happen in a few different places.